hypershift-operator/controllers/nodepool/haproxy.go

```diff
@@ -121,13 +121,15 @@ func (r *NodePoolReconciler) reconcileHAProxyIgnitionConfig(ctx context.Context,
 		}
 	}
 	var apiserverProxy string
+	var noProxy string
 	if hcluster.Spec.Configuration != nil && hcluster.Spec.Configuration.Proxy != nil && hcluster.Spec.Configuration.Proxy.HTTPSProxy != "" && util.ConnectsThroughInternetToControlplane(hcluster.Spec.Platform) {
 		apiserverProxy = hcluster.Spec.Configuration.Proxy.HTTPSProxy
+		noProxy = hcluster.Spec.Configuration.Proxy.NoProxy
 	}
 
 	machineConfig := manifests.MachineConfigAPIServerHAProxy()
 	ignition.SetMachineConfigLabels(machineConfig)
-	serializedConfig, err := apiServerProxyConfig(haProxyImage, controlPlaneOperatorImage, apiServerExternalAddress, apiServerInternalAddress, apiServerExternalPort, apiServerInternalPort, apiserverProxy)
+	serializedConfig, err := apiServerProxyConfig(haProxyImage, controlPlaneOperatorImage, apiServerExternalAddress, apiServerInternalAddress, apiServerExternalPort, apiServerInternalPort, apiserverProxy, noProxy, hcluster.Spec.Networking.ServiceNetwork[0].CIDR.String(), hcluster.Spec.Networking.ClusterNetwork[0].CIDR.String())
```

Comment:
Where are we checking that `hcluster.Spec.Networking.ServiceNetwork` is not zero-length?

Reply:
This is defaulted in the API, I can still add a safety check